tls/s2n_config.c

```diff
@@ -938,3 +938,11 @@ int s2n_config_client_hello_cb_enable_poll(struct s2n_config *config) {
 
     return S2N_SUCCESS;
 }
+
+int s2n_config_set_send_buffer_size(struct s2n_config *config, uint32_t buffer_byte_size) {
+    POSIX_ENSURE_REF(config);
+
+    config->send_buffer_size = buffer_byte_size;
```

Comment:
Should there be a minimum value? Like if they set it to `1` we wouldn't even be able to write a record header.

Reply:
Ah great point. I think the minimum value should be the TLS max record size + record overhead because we cannot have a smaller buffer until after the handshake, making this API a lot more complicated.

Thoughts?